GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -465,21 +465,22 @@ List<TreeNode> SearchTree(string text, IEnumerable<TreeNode> nodes)
             {
                 Queue<TreeNode> queue = new(nodes);
                 List<TreeNode> ret = new();
-
+                bool isFullMatch = false;
+                int matchIndex = -1;
                 while (queue.Count != 0)
                 {
                     var n = queue.Dequeue();
 
                     if (n.Tag is BaseBranchNode branch)
                     {
-                        if (branch.FullPath.IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1)
+                        if (searchForMatchingText(branch.FullPath, text, ref isFullMatch, ref matchIndex, ret))
```

Comment:
Still duplicated code
```suggestion
                    string nodeText = n.Tag is BaseBranchNode branch ? branch.FullPath : n.Text;
                    if (searchForMatchingText(nodeText, text, ref isFullMatch, ref matchIndex, ret))
                    {
                        AddTreeNodeToSearchResult(ret, n);
                    }
```